src/query/interpret/eval.hpp

```diff
@@ -261,6 +260,20 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
     }
   }
 
+  TypedValue Visit(OrOperator &op) override {
+    auto value1 = op.expression1_->Accept(*this);
+    if (value1.IsBool() && value1.ValueBool()) {
+      // If first expression is true, don't evaluate the second one.
+      return value1;
+    }
+    auto value2 = op.expression2_->Accept(*this);
+    try {
+      return value1 || value2;
```

Comment:
if 
```suggestion
      if (value2.IsBool()) return value2;
      return false;
```

Reply:
if we came to this point, value1 is not bool or false so I think correct thing is just to return value2 ?